dotnet/src/SemanticKernel.UnitTests/Functions/KernelFunctionMetadataFactoryTests.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.ComponentModel;
+using System.Linq;
+using Microsoft.SemanticKernel;
+using Xunit;
+
+namespace SemanticKernel.UnitTests.Functions;
+public class KernelFunctionMetadataFactoryTests
+{
+    [Fact]
+    public void ItCanCreateFromType()
+    {
+        // Arrange
+        var instanceType = typeof(MyKernelFunctions);
+
+        // Act
+        var functionMetadata = KernelFunctionMetadataFactory.CreateFromType(instanceType);
+
+        // Assert
+        Assert.NotNull(functionMetadata);
+        Assert.Equal(3, functionMetadata.Count<KernelFunctionMetadata>());
+        Assert.Contains(functionMetadata, f => f.Name == "Function1");
+        Assert.Contains(functionMetadata, f => f.Name == "Function2");
+        Assert.Contains(functionMetadata, f => f.Name == "Function3");
+    }
+
+    #region private
+#pragma warning disable CA1812 // Used in test case above
+    private sealed class MyKernelFunctions
+    {
+        // Disallow instantiation of this class.
+        private MyKernelFunctions()
+        {
+        }
```

Comment:
No constructor needed.

```suggestion
```

Reply:
Having a private constructor is part of the test, I want to ensure the code doesn't try to create an instance of the class